src/norm_flows.jl

```diff
@@ -0,0 +1,98 @@
+using Distributions
+using LinearAlgebra
+using Random
+using StatsFuns: softplus
+using Roots, LinearAlgebra # for inverse
+
+################################################################################
+#                            Planar and Radial Flows                           #
+#             Ref: Variational Inference with Normalizing Flows,               #
+#               D. Rezende, S. Mohamed(2015) arXiv:1505.05770                  #
+################################################################################
+
+mutable struct PlanarLayer <: Bijector
+    w
+    u
+    u_hat
+    b
+end
+
+mutable struct RadialLayer <: Bijector
+    α_
+    β
+    z_not
+end
+
+function update_u_hat(u, w)
+    # to preserve invertibility
+    u_hat = u + (planar_flow_m(transpose(w) * u) - transpose(w) * u)[1] * w/(norm(w[:,1],2) ^ 2)
+end
+
+function update_u_hat!(flow::PlanarLayer)
+    flow.u_hat = flow.u + (planar_flow_m(transpose(flow.w) * flow.u) - transpose(flow.w) * flow.u)[1] * flow.w/(norm(flow.w,2)^2)
```

Comment:
I was thinking this might be the case, but when using `Tracker` wouldn't this turn a `TrackedArray` into an array of `TrackedReal`?

With the current approach you still maintain the `TrackedArray`, which ought to be more efficient.